internal/server/data/data.go

```diff
@@ -152,28 +139,62 @@ func add[T models.Modelable](db *gorm.DB, model *T) error {
 		return err
 	}
 
-	if err := db.Create(model).Error; err != nil {
-		if isUniqueConstraintViolation(err) {
-			return fmt.Errorf("%w: %s", internal.ErrDuplicate, err)
-		}
+	err := db.Create(model).Error
+	return handleError(err)
+}
 
-		return err
-	}
+type UniqueConstraintError struct {
```

Comment:
This should ideally store table, column, and value but some implementations (sqlite3) does not report the value. Even in cases where it does report the value, the value may be an ID which wouldn't be very useful to the end user